devito/types/dimension.py

```diff
@@ -445,6 +449,11 @@ def spacing(self):
     def _defines(self):
         return frozenset({self}) | self.parent._defines
 
+    @cached_property
+    def level(self):
+        # The depth in the hierarchy of `IncrDimension`s
+        return len([i for i in self._defines if i.is_Incr])
```

Comment:
No, this is wrong. You don't need to search for `is_Incr` ... And it is wrong twice, because you're searching for a subclass in a superclass, and superclasses don't see subclasses. 

In fact, maybe we don't even need to override? how about just 

```
class Dimension(...)
     @property
     def level(self):
          return len(self._defines)
```

is this enough?

Reply:
Subdim blocking e.g. lots of tests in TestSubDimension. I guess obvious why.

For non subdims tests:

This is enough because in space blocking you do relative level comparison. So the returned value will be altered now but still valid as it is gonna be compared with another altered value.
SO like instead of having  
3>=2
in e.g.
```python
if i.dim.level >= bd.level:
```
we now gonna have
4>=3

Still valid, but different valued hierarchy

Reply:
Then we just the property from Derived to Incr  as before?

Reply:
> . I guess obvious why.

can u elaborate pls?

Reply:
>. I guess obvious why.
> can u elaborate pls?

So there is not problem.
I apologize. It breaks only with skewing options, as the `relations` strategy is different. Will fix and recommit!

Reply:
So the issue is that in master skewing I need to know the absolute position in Incr-only hierarchy... the rest as I wrote on slack